### DISCUSSION TITLE
Implement "OR" and "AND" conditionals via an `if... then` section

### DISCUSSION THREAD

**Discussion by abandonedaccount6235** [SkriptLang/Skript#5152] [Suggestions] [answered]
**Problem**
A long standing annoyance in Skript is that conditionals can only accept a single condition. Due to how Skript is designed and written, it's not very practical to implement traditional `&&`/`||` type operators. This makes creating "and"/"or" conditionals cumbersome for scripters, as you must either indent a ton of levels (for "and"):
```
if foo() is bar():
  if otherfoo() is otherbar():
    if otherotherfoo() is otherotherbar():
      # and so on...
```
or you must set vars to check (for "or"):
```vb
set {_or} to true if foo() is bar()
set {_or} to true if otherfoo() is otherbar()
set {_or} to true if otherotherfoo() is otherotherbar()
if {_or} is true:
  # do something
```
While you could use inline conditions to make `and` slightly less annoying to write, you are then unable to use an "else".

**Proposed Solution**
Implement a `if... then` scope for "and"/"or" conditions. For example, the above would be written as:
```
if:
  foo() is bar()
  otherfoo() is otherbar()
  otherotherfoo() is otherotherbar()
then:
  # do something if all the conditions were true
```
```
if at least one of:
  foo() is bar()
  otherfoo() is otherbar()
  otherotherfoo() is otherotherbar()
then:
  # do something if at least one of the conditions was true
```

As you can see, these are both way cleaner and easier to write than the original code. In addition, the parser can handle it better as each condition is on it's own line. It even increases performance as the "or" example can now be short circuited behind the scenes!

**Proposed Syntax**
"or" syntax: `if (any|at least one [of]):`
"and" syntax: `if [all]:`
"then" syntax: `then [(run|perform|do)]`

**Other Considerations**

- Once conditions are boolean expression instead, this may no longer be necessary
  - In my opinion, chaining conditions will likely still be an issue after this change. The parser is still prone to slowing down on long lines, and even if it wasn't I think it would still often parse things in unintended ways when chaining lots of `or`/`and` operators.

**Comment by Ankoki** [comment]
I love this idea, i think it would be good to make sure an `else` section underneath the `then` work as well, just something to add just in case:) But overall this is an amazing idea,

**Comment by abandonedaccount6235** [comment, marked as answer]
Feature has been merged in #5165!
